test/model/src/core/kv_table.cpp

```diff
@@ -227,6 +227,59 @@ kv_table::remove(kv_transaction_ptr txn, const data_value &key)
     }
 }
 
+/*
+ * kv_table::truncate --
+ *     Truncate a key range.
+ */
+int
+kv_table::truncate(const data_value &start, const data_value &stop, timestamp_t timestamp)
+{
+    std::lock_guard lock_guard(_lock);
+    if (start != model::NONE && stop != model::NONE && start > stop)
+        throw model_exception("The start and the stop key are not in the right order");
+
+    auto start_iter = start == model::NONE ? _data.begin() : _data.lower_bound(start);
+    auto stop_iter = stop == model::NONE ? _data.end() : _data.upper_bound(stop);
+
+    try {
+        for (auto i = start_iter; i != stop_iter; i++)
+            i->second.add_update(
+              std::move(kv_update(NONE, fix_timestamp(timestamp))), false, false);
+    } catch (wiredtiger_exception &e) {
+        return e.error();
+    }
+
+    return 0;
+}
+
+/*
+ * kv_table::truncate --
+ *     Truncate a key range.
+ */
+int
+kv_table::truncate(kv_transaction_ptr txn, const data_value &start, const data_value &stop)
+{
+    std::lock_guard lock_guard(_lock);
+    if (start != model::NONE && stop != model::NONE && start > stop)
+        throw model_exception("The start and the stop key are not in the right order");
+
+    auto start_iter = start == model::NONE ? _data.begin() : _data.lower_bound(start);
+    auto stop_iter = stop == model::NONE ? _data.end() : _data.upper_bound(stop);
+
+    try {
+        for (auto i = start_iter; i != stop_iter; i++) {
+            std::shared_ptr<kv_update> update =
```

Comment:
Would it be ok to update the other function to take an update as an argument and call it like:
```
truncate(update, start, stop)
```
Or is it weird to have a truncate function that takes an update? It could be fine if it is an internal function.

Reply:
An update is currently designed to be unique per key, as it will have more fields filled in when it is applied. (It is also an internal object, which may not make much sense to expose.)